authorization/invitation/service/invitation_service.go

```diff
@@ -0,0 +1,154 @@
+	repo    repository.Repositories
+	permSvc permissionService.PermissionService
+}
+
+func NewInvitationService(repositories repository.Repositories) InvitationService {
+	return &InvitationServiceImpl{repo: repositories, permSvc: repositories.PermissionService()}
+}
+
+// Issue creates new invitations.  The inviteTo parameter is the unique id of the organization, team, security group or resource for
+// which the invitations will be issued, and the invitations parameter contains the users and state for each individual user invitation.
+// This method creates one record in the INVITATION table for each user in the invitations parameter.  Any roles that are issued
+// as part of a user's invitation are created in the INVITATION_ROLE table.
+func (s *InvitationServiceImpl) Issue(ctx context.Context, issuingUserId uuid.UUID, inviteTo string, invitations []invitation.Invitation) error {
+	var inviteToIdentity *account.Identity
+	var identityResource *resourceRepo.Resource
+	var inviteToResource *resourceRepo.Resource
+
+	// First try to convert inviteTo to a uuid
+	inviteToUUID, err := uuid.FromString(inviteTo)
+	// If we get an error here, the value is definitely not for an Identity so we'll treat it as a resource ID
+	if err != nil {
+		// Try to lookup a resource with the same ID value
+		inviteToResource, err = s.repo.ResourceRepository().Load(ctx, inviteTo)
+		if err != nil {
+			return errors.NewNotFoundError(fmt.Sprintf("invalid identifier '%s' provided for organization, team, security group or resource", inviteTo), inviteTo)
+		}
+	}
+
+	// If we didn't successfully find a valid resource already, it means the inviteTo is a UUID
+	if inviteToResource == nil {
+		// Attempt to lookup the identity of the organization, team or security group that invitations will be issued for
+		inviteToIdentity, err = s.repo.Identities().Load(ctx, inviteToUUID)
+		if err != nil {
+			// That didn't work, try to lookup a resource with the same ID value
+			inviteToResource, err = s.repo.ResourceRepository().Load(ctx, inviteTo)
+			if err != nil {
+				return errors.NewNotFoundError(fmt.Sprintf("invalid identifier '%s' provided for organization, team, security group or resource", inviteTo), inviteTo)
+			}
+		}
+	}
+
+	if inviteToIdentity != nil {
+		// Load the resource for the identity
+		identityResource, err = s.repo.ResourceRepository().Load(ctx, *inviteToIdentity.IdentityResourceID)
+		if err != nil {
+			return errors.NewInternalErrorFromString(ctx, "error loading resource for identity")
+		}
+
+		// Confirm that the issuing user has the necessary scope to manage members for the organization, team or security group
+		scope, err := s.permSvc.HasScope(ctx, issuingUserId, *inviteToIdentity.IdentityResourceID, authorization.ManageMembersScope)
+		if err != nil {
+			return errors.NewInternalError(ctx, err)
+		}
+
+		if !scope {
+			return errors.NewForbiddenError(fmt.Sprintf("user requires %s scope to invite other users", authorization.ManageMembersScope))
```

Comment:
This is very important use case (when the user don't have permissions to invite) but it's not covered by tests - https://codecov.io/gh/fabric8-services/fabric8-auth/pull/355/diff#D28-79

Reply:
Test added.